### PR TITLE
Natural Language Processing using Gradient Boosting Machine and H2O library

### LINKED ISSUE #6171: [Machine Learning] Natural language processing using Gradient Boosting Machine and H2O library
#### Proposed title of article
[Machine Learning] Natural language processing using Gradient Boosting Machine(GBM) and H2O library

#### Proposed article introduction
Natural language processing helps machines understand and respond to text or voice data like human beings. We will use the Gradient Boosting Machine and H2O library to build a natural language processing model. GBM is an algorithm that combines the predictions from multiple decision trees to generate the final predictions. Every successive decision tree is built on the errors of the previous trees. The trees in GBM are sequentially and the final output is a model that is fully optimized.  GBM converts weak learners into strong learners.

H2O's is used for automating the machine learning workflow, which includes automatic training and tuning of many models within a user-specified time limit. H2O offers a number of model explainability methods that apply to AutoML objects. This speeds up the development process and saves developers time. H2O also applies various optimization techniques in choosing the best algorithm to build the model. 

Using two libraries we will have every effective model. The model will have high performance and will be able to make accurate predictions.

#### Key takeaways 
1.  Gradient Boosting Machine.
2. Text preprocessing activities.
3. Installing H2O library.
4. H20 methods and functions.
5. Building the model for natural language processing using GBM and H2O.

#### Article quality
This article is unique because we will be building a custom model. We will build a sentiment analysis model which is a type of natural language processing. We will then explore the various GBM method to give a reader a better understanding. Finally, we will integrate GBM and H2O so that we have a fully optimized model. The model will have a high-performance accuracy score and make accurate predictions.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by willyngashu**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6171

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by LinusMuema** [comment]
plagiarism check done ✔️

File: content/articles/natural-language-processing-using-gradient-boosting-machine-and-h2o-library/index.md [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+author: willyngashu
+date: 2022-02-10T00:00:00-21:08
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/natural-language-processing-using-gradient-boosting-machine-and-h2o-library/hero.png
+    alt: Natural Language Processing using Gradient Boosting Machine and H2O library Hero image
+---
+H2O is an open-source platform used for machine learning. H2O supports the implementation of popular machine learning algorithms. H20 supports both supervised and unsupervised machine learning algorithms.
```
Comment (Bayler) [inline content/articles/natural-language-processing-using-gradient-boosting-machine-and-h2o-library/index.md:17]: Add a external link to H2O

**Comment by Bayler** [comment]
@willyngashu Did you perform a grammarly check on your tutorial? I can see lots of grammatical mistakes

**Comment by willyngashu** [comment]
Hello @Bayler, I have gone through the article and made the requested changes.
Thanks.

**Comment by Bayler** [comment]
@willyngashu Have you used [Grammarly](https://app.grammarly.com/) to help you perform the check? I can still see the same grammatical mistakes.

**Comment by willyngashu** [comment]
Hello @Bayler, I have used [Grammarly](https://app.grammarly.com/) to perform the check. https://app.grammarly.com/ddocs/1482426025. If you feel I have not implemented the changes, do you mind highlighting the mistakes? 
Thanks.
Screenshot (96)

**Comment by Bayler** [comment]
@willyngashu Please take note of all the changes made in this tutorial so that you don't make the same mistakes in your future tutorials. As it is, it is ready for final review.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**